### PR TITLE
METRON-1441: Create complementary Solr schemas for the main sensors

### PR DESCRIPTION
**Pull request by cestella**
## Contributor Comments
We have ES templates for bro, snort, yaf, and error, we need corresponding solr schemas for these collections.

Right now this is tested via the SchemaTranslatorTest, which spins up the current version of Solr loads the schemas and writes sample data and ensures they come back properly. 

Note: This is an intermediate move to getting full solr support.  These schemas will be refined more, likely.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
...ver\/BrokerTopicMetrics\/AllTopicsBytesOutPerSec\/1MinuteRate[1484165330,1484168930,15],metrics\/kafka\/server\/BrokerTopicMetrics\/AllTopicsMessagesInPerSec\/1MinuteRate[1484165330,1484168930,15],metrics\/kafka\/controller\/KafkaController\/ActiveControllerCount[1484165330,1484168930,15],metrics\/kafka\/controller\/ControllerStats\/LeaderElectionRateAndTimeMs\/1MinuteRate[1484165330,1484168930,15],metrics\/kafka\/controller\/ControllerStats\/UncleanLeaderElectionsPerSec\/1MinuteRate[1484165330,1484168930,15],metrics\/kafka\/server\/ReplicaFetcherManager\/Replica-MaxLag[1484165330,1484168930,15],metrics\/kafka\/server\/ReplicaManager\/PartitionCount[1484165330,1484168930,15],metrics\/kafka\/server\/ReplicaManager\/UnderReplicatedPartitions[1484165330,1484168930,15],metrics\/kafka\/server\/ReplicaManager\/LeaderCount[1484165330,1484168930,15]&format=null_padding&_=1484168930776","tags":[],"source.type":"bro","adapter.geoadapter.end.ts":"1517499201209","referrer":"http:\/\/node1:8080\/","threatintelsplitterbolt.splitter.end.ts":"1517499201215","adapter.threatinteladapter.begin.ts":"1517499201217","ip_src_port":50451,"guid":"b62fe444-82fb-46a4-8c4a-5cfc248bee41","response_body_len":0}
```
Comment (ottobackwards) [inline metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java]: Can we make these files please?  Maybe add a new directory "enriched" under the sample data dir in the integration testing module

File: metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+import org.apache.metron.solr.writer.SolrWriter;
+import org.json.simple.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.*;
+
```
Comment (ottobackwards) [inline metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java]: Shouldn't this be an integration test?

Comment (cestella) [reply metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java]: Yes, I absolutely can.

File: metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java (hunk 1 above)
Comment (cestella) [reply metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java]: We could, though this test probably won't last past the feature branch.

**Comment by cestella** [comment]
@ottobackwards Very likely these schema files won't stay in this spot, but the final resting spot won't be apparent until we figure out how to automatically apply the schemas.  Treat this PR as just unlocking progress for downstream PRs (like correcting SolrWriter to write to Solr again).

File: metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java (hunk 1 above)
Comment (ottobackwards) [reply metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java]: I wouldn't be so sure.  When we get serious about ES templates and these schemas they will be tested per parser.   As part of developing a parser you can validate your schemas etc like this.

Comment (cestella) [reply metron-platform/metron-solr/src/test/java/org/apache/metron/solr/schema/SchemaTranslatorTest.java]: Alright, you got it then :)

**Comment by cestella** [comment]
Ok, I did the following:
* Augmented the readme to point to the Solr documentation around schemas.  Keep in mind, this is intermediate work that will feed into the "install Solr" work
* Added yaf and error schemas
* Renamed the test to an integration test
* Moved the data from multiline to separate files

**Comment by merrimanr** [comment]
I tested this in full dev using the install script in https://github.com/apache/metron/pull/918.  I was able to create collections for each schema except for "error".  For that to work properly, I had to: 

- remove `docValues="true"` from the "bytes" field type
- add the "guid" field used in other schemas

Still working on indexing data into these collections but so far so good.

**Comment by cestella** [comment]
@merrimanr I'm confused about the guid field.  I had to take that out given the sample data that I generated from full-dev (they didn't have a `guid` field).  How did you create the errors?

**Comment by merrimanr** [comment]
The errors I found happened when I tried to create collections with each schema.  The error related to the `guid` field happens because it is defined as the unique key but not included in the list of fields (for error schema).

**Comment by cestella** [comment]
@merrimanr Ah!  Yes, we really should set a `guid` field for errors.

**Comment by cestella** [comment]
@merrimanr Things should be fixed at this point.  I'm going to spin it up and validate things this morning.

**Comment by ottobackwards** [comment]
So, the idea is that we have a 'core' for each parser?  and for errors?

**Comment by merrimanr** [comment]
I would say collection for each parser and an error collection.  It's similar to ES templates where we define fields, types, etc.

**Comment by cestella** [comment]
Ugh, sorry, missed the guid field.  *now* it's ready to test with guid field set up.

**Comment by merrimanr** [comment]
Looks good to me.  +1

**Comment by simonellistonball** [comment]
Should we tidy up the ordering of the schema files for better legibility (uniquekey next to the field, dynamic catch alls in a consistent location, some semantic ordering of the key elements)?

**Comment by cestella** [comment]
@simonellistonball Yes, we should.  I added the relevant context and grouping for each of the schemas that we ship by default.  Bro has more context as there were more comments in the ES schema.  I brought over whatever comments were relevant and grouped appropriately and as best I could in the others.

Let me know if I missed anything.

File: metron-platform/metron-solr/src/main/config/schema/error/solrconfig.xml [hunk 3]
```diff
@@ -0,0 +1,1601 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (simonellistonball) [inline metron-platform/metron-solr/src/main/config/schema/error/solrconfig.xml]: Is this just basic stock solr? The /browse endpoint configs seem like they come from the tutorial sample, and could be mis-leading for example.

**Comment by simonellistonball** [comment]
@cestella much neater. Thank you! I'll put my data schema OCD away now.

**Pull request closed without merging**